recipes/leveldb/all/conanfile.py

```diff
@@ -0,0 +1,84 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class LevelDBConan(ConanFile):
+    name = "leveldb"
+    description = ("LevelDB is a fast key-value storage library written at "
+                   "Google that provides an ordered mapping from string keys "
+                   "to string values")
+    license = "https://github.com/google/leveldb/blob/master/LICENSE"
+    topics = ("conan", "leveldb", "google", "db")
+    url = "https://github.com/conan-io/conan-center-index "
+    homepage = "https://github.com/google/leveldb"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+            "shared": [True, False],
+            "fPIC": [True, False],
+            "with_snappy": [True, False],
+    }
+    default_options = {
+            "shared": False,
+            "fPIC": False,
+            "with_snappy": False,
+    }
+    generators = "cmake"
+
+    _cmake = None
+
+    def _get_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["LEVELDB_BUILD_TESTS"] = False
+        if self.options.fPIC:
+            self._cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = True
+        return self._cmake
+
+    # note: crc32, tcmalloc are also conditionally included in leveldb, but
+    # there are no "official" conan packages yet; when those are available, we
+    # can add similar with- options for those
+    optional_snappy_requirement = "snappy/1.1.7"
+
+    def requirements(self):
+        if self.options.with_snappy:
+            self.requires(self.optional_snappy_requirement)
```

Comment:
```suggestion
    # FIXME: crc32, tcmalloc are also conditionally included in leveldb, but
    # there are no "official" conan packages yet; when those are available, we
    # can add similar with options for those

    def requirements(self):
        if self.options.with_snappy:
            self.requires("snappy/1.1.7")
```

You can create these options, with default value `False`.
When that option is enabled, you then raise an `ConanInvalidConfiguration` exception in `requirements`.
Don't forget to add a `#FIXME`.